### PR TITLE
Make ICommandPalette, IMainMenu and ILayoutRestorer optional

### PR DESCRIPTION
**Pull request by jtpio**
So the plugins can be more easily reused elsewhere (https://github.com/TheSnakePit/mamba-navigator/issues/8)

### CODE REVIEW

File: labextension/src/index.ts [hunk 1]
```diff
@@ -30,10 +30,10 @@ const TOUR_TIMEOUT = 5 * TOUR_DELAY + 1;
 
 async function activateCondaEnv(
   app: JupyterFrontEnd,
-  palette: ICommandPalette,
-  menu: IMainMenu,
-  restorer: ILayoutRestorer,
-  settingsRegistry: ISettingRegistry
+  settingsRegistry: ISettingRegistry,
```
Comment (jtpio) [inline labextension/src/index.ts:33]: It looks like the settings registry could also be made optional (and have `CondaEnvironments` use a set of defaults)

But this would be fine to do in a separate PR.

Comment (jtpio) [reply labextension/src/index.ts:33]: The settings registry is actually optional here:

https://github.com/fcollonval/jupyter_conda/blob/a7fd182a6b6a47a7ca48b545ba946e2937fcf4b1/labextension/src/services.ts#L61

So we could indeed make it optional at the plugin level too.

**Comment by fcollonval** [comment]
Thanks for starting this @jtpio 

The question of the settings is a good one. We could indeed make it optional. But I'm pretty sure settings will come handy at some point for the navigator, too. The question is how to handle them in mamba-navigator; from the backend, reusing @jupyterlab/settings, something else...

**Comment by fcollonval** [comment]
> sidenote: The windows job has the bad habit of failing on `jupyterlab.browser_check` step.

**Comment by jtpio** [comment]
The one for `jupyter_conda:plugin` can be made optional without changing too much of the logic: https://github.com/fcollonval/jupyter_conda/pull/73/commits/9b751de8566522d606effef9055e6f83223a444c

Then yes the settings will probably be useful for the mamba navigator too, although it's not decided yet if they will be exposed to the end user. If they are they could also be edited with an alternative or custom editor widget.

But it looks like the logic in https://github.com/fcollonval/jupyter_conda/blob/master/labextension/src/validator.ts should work fine even if the settings are not available, with a predefined set of `companions` is instead.

**Comment by fcollonval** [comment]
Thanks @jtpio I need to think a bit more about the validator.
